Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -710,6 +710,36 @@ CGImageRef _CGImageLoadImageWithWICDecoder(REFGUID decoderCls, void* bytes, int
     return nil;
 }
 
+size_t _CGImageImputeBitsPerPixelFromFormat(CGColorSpaceRef colorSpace, size_t bitsPerComponent, CGBitmapInfo bitmapInfo) {
+    unsigned int alphaInfo = bitmapInfo & kCGBitmapAlphaInfoMask;
+    unsigned int byteOrder = bitmapInfo & kCGBitmapByteOrderMask;
+
+    // Try byte order first: The user can specify 32 or 16 directly.
+    switch (byteOrder) {
+        case kCGBitmapByteOrder32Little:
+        case kCGBitmapByteOrder32Big:
+            return 32;
+        case kCGBitmapByteOrder16Little:
+        case kCGBitmapByteOrder16Big:
+            return 16;
```

Comment:
Given that we don't support 16 bit contexts, should we failfast or early error here?